web/scripts/users/profileComponent.js

```diff
@@ -9,40 +9,69 @@ module.exports = {
+            <label>Full Name</label></br>
+            {{storage.get('user').name}}
+          </div>
+          <div class="form-group">
+            <label>Email</label></br>
+            {{storage.get('user').email}}
+          </div>
+          </br>
+          <div class="form-group">
+            <label>Categories</label></br>
+            <table class="category-table">
+              <tr v-for="cat in storage.get('user').categories">
+                <td class="td">
+                  {{cat}}
+                </td>
+                <td class = "td">
+                <button class="btn btn btn-danger btn-sm" v-on:click="removeCat(cat)">Remove</button>
+                </td>
+              </tr>
+              <tr>
+                <td class = "td"><input v-model="newcat" name="newcat" class="form-control form-control-sm" type="text" placeholder="New Category"></td>
+                <td class = "td">
+                <button class="btn btn btn-success btn-sm" v-on:click="addCat">Add</button>
+                </td>
+              </tr>
+            </table>
+          </div>
+          <button class="btn btn-primary" v-on:click="logout">Logout</button>
+          <br />
+          <br />
         </div>
-        <div class="form-group">
-          <label>Email</label></br>
-          {{storage.get('user').email}}
-        </div>
-        </br>
-        <div class="form-group">
-          <label>Categories</label></br>
-          <table class="category-table">
-            <tr v-for="cat in storage.get('user').categories">
-              <td class="td">
-                {{cat}}
-              </td>
-              <td class = "td">
-              <button class="btn btn btn-danger btn-sm" v-on:click="removeCat(cat)">Remove</button>
-              </td>
-            </tr>
-            <tr>
-              <td class = "td"><input v-model="newcat" name="newcat" class="form-control form-control-sm" type="text" placeholder="New Category"></td>
-              <td class = "td">
-              <button class="btn btn btn-success btn-sm" v-on:click="addCat">Add</button>
-              </td>
-            </tr>
-          </table>
+
+        <div id="purchase-history">
+          <h1>Purchase History</h1>
+          <div class="purchase-item" v-for="purchase in storage.get('purchaseHistory')">
+            <label>{{moment(purchase.time).format('MMMM Do YYYY')}}</label></br>
+            <label>Total:</label>&nbsp{{'$' + purchase.amount/100}}
+            </br>
+            <table v-if="purchase.items.length >= 1">
+              <col width="60%">
+              <col width="10%">
+              <tr>
+                <th class="pitem">Item</th>
+                <th class="pprice">Price</th>
+                <th class="pauthor">Author(s)</th>
+              </tr>
+              <tr v-for="item in purchase.items">
+                  <td>
+                    <a href="javascript:void(0);" v-on:click="viewItem(item)">{{item.title}}</a>
```

Comment:
What do you think about instead of the title bringing them to the book page the link instead downloads a dummy pdf for them? So they click on the title an alert box pops up asking if they'd like to download this e-book. If they click yes, then we grab a free e-book from some site and it gets downloaded onto their computer. I believe Andy mentioned he'd like this kind of functionality and it should be pretty easy since it can be the same pdf file for each link

Reply:
This may prove to be a good functionality. But I think that it is still important to have a link to direct user to the book page. Perhaps we can include another column with a small icon for that?

Reply:
👍